packages/nuxt-module/plugins/domain.js

```diff
@@ -0,0 +1,92 @@
+import Middleware from "./middleware";
+import VueCompositionAPI, { computed, ref } from "@vue/composition-api";
+import { useSessionContext } from "@shopware-pwa/composables";
+import Vue from "vue";
+const FALLBACK_DOMAIN = "<%= options.fallbackDomain %>" || "/";
+const FALLBACK_LOCALE = "<%= options.fallbackLocale %>";
+const PWA_HOST = "<%= options.pwaHost %>";
+const domainsList = require("sw-plugins/domains");
+Vue.use(VueCompositionAPI);
+
+const currentDomainData = ref();
+// register domains based routing and configuration
+export default ({ app, route }, inject) => {
+  const routeDomainUrl = computed(() => route.meta?.[0]?.url);
+  const routeDomain = computed(() =>
+    Object.values(domainsList).find(
+      (domain) => domain.url === routeDomainUrl.value
+    )
+  );
+
+  const routing = {
+    // list of available domains from "domains.json" - output of "domains" CLI command
+    availableDomains: Object.values(domainsList),
+    fallbackDomain: FALLBACK_DOMAIN,
+    fallbackLocale: FALLBACK_LOCALE,
+    pwaHost: PWA_HOST,
+    // set current domain's configuration
+    setCurrentDomain: (domainData) => {
+      currentDomainData.value = domainData;
+    },
+    // get current domain's configuration
+    getCurrentDomain: () => {
+      return currentDomainData.value || routeDomain.value;
+    },
```

Comment:
this one should be computed properties, not methods
```suggestion
    getCurrentDomain: computed(() => {
      return currentDomainData.value || routeDomain.value;
    }),
```